backend/src/main/java/com/agh/introwertycznelosie/data/Faculty.java

```diff
@@ -1,21 +1,118 @@
 package com.agh.introwertycznelosie.data;
 
-public enum Faculty {
-
-    WGiG,
-    WIMIP,
-    WEAiIB,
-    WIEiT,
-    WIMiR,
-    WGGiOS,
-    WGGiIS,
-    WIMiC,
-    WO,
-    WMN,
-    WWNiG,
-    WZ,
-    WEiP,
-    WFiIS,
-    WMS,
-    WH
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+@JsonSerialize(using = FacultySerializer.class)
+@JsonDeserialize(using = FacultyDeserializer.class)
+public class Faculty {
+
+    public class InvalidFacultyException extends Exception {
+        public InvalidFacultyException(String errorMessage) {
+            super(errorMessage);
+        }
+    }
+
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private String name;
+    private String acronym;
+
+    public Faculty(){}
+
+    public Faculty(String name, String acronym){
+        this.name = name;
+        this.acronym = acronym;
+    }
+    //TODO change this method after adding dynamic Faculties on frontend
+    public Faculty(String acronym) throws InvalidFacultyException {
+        this.acronym = acronym;
```

Comment:
Ja bym to dała pod switchem, bo jeśli będzie błędny akronim to podstawi do głównego pola źle zanim wyrzuci wyjątek